staging/src/k8s.io/kubectl/pkg/cmd/diff/diff_test.go

```diff
@@ -636,3 +642,35 @@ func TestMasker(t *testing.T) {
 		})
 	}
 }
+
+// TODO - This is still not working, need to understand diff code better
+func TestValidateFields(t *testing.T) {
+	tests := []struct {
+		args         [][]string
+		enableAlphas []cmdutil.FeatureGate
+		expectedErr  string
+	}{
+		{
+			args: [][]string{
+				{"server-side", "auto"},
+				{"force-conflicts", "true"},
+			},
+			enableAlphas: []cmdutil.FeatureGate{cmdutil.AutoServerSide},
+			expectedErr:  "--force-conflicts only works with --server-side",
+		},
+	}
+	for i, test := range tests {
+		t.Run(fmt.Sprintf("case %d", i), func(t *testing.T) {
```

Comment:
So I dug into diff, and because it hasn't separated flags from options (and exposes neither from NewCmdDiff), I think we need to use the following pattern, which I see in some apply tests:

```diff
@@ -664,12 +666,19 @@ func TestValidateFields(t *testing.T) {
                        f := cmdtesting.NewTestFactory()
                        defer f.Cleanup()
                        cmdtesting.WithAlphaEnvs(test.enableAlphas, t, func(t *testing.T) {
+                               cmdutil.BehaviorOnFatal(func(str string, code int) {
+                                       panic(str)
+                               })
+                               defer func() {
+                                       actualError := recover()
+                                       require.Equal(t, test.expectedErr, actualError)
+                               }()
                                cmd := NewCmdDiff(f, genericclioptions.NewTestIOStreamsDiscard())
                                cmd.Flags().Set("filename", filenameRC)
                                for _, arg := range test.args {
                                        cmd.Flags().Set(arg[0], arg[1])
                                }
-
+                               cmd.Run(cmd, []string{})
                        })
                })
        }
```